version: "3.7"
services:
  django_app:
    image: django_app:latest
    volumes:
      - django_app_data:/app
    networks:
      - network_public
    environment:
      - DJANGO_DEBUG=False
      - DJANGO_ALLOWED_HOSTS=appteste.mibitech.com.br
      - VIRTUAL_HOST=appteste.mibitech.com.br
    deploy:
      mode: replicated
      replicas: 1
      placement:
        constraints:
          - node.role == manager
      labels:
        - traefik.enable=true
        - traefik.http.routers.django_app.rule=Host(`appteste.mibitech.com.br`)
        - traefik.http.routers.django_app.entrypoints=websecure
        - traefik.http.routers.django_app.tls.certresolver=letsencryptresolver
        - traefik.http.routers.django_app.service=django_app
        - traefik.http.services.django_app.loadbalancer.server.port=8000
        - traefik.http.services.django_app.loadbalancer.passHostHeader=true

  nginx:
    image: nginx_app:latest
    volumes:
      - nginx_data:/var/log/nginx
    networks:
      - network_public
    deploy:
      mode: replicated
      replicas: 1
      placement:
        constraints:
          - node.role == manager
      labels:
        - traefik.enable=true
        - traefik.http.routers.nginx_app.rule=Host(`nginx-appteste.mibitech.com.br`)
        - traefik.http.routers.nginx_app.entrypoints=websecure
        - traefik.http.routers.nginx_app.tls.certresolver=letsencryptresolver
        - traefik.http.routers.nginx_app.service=nginx_app
        - traefik.http.services.nginx_app.loadbalancer.server.port=80
        - traefik.http.services.nginx_app.loadbalancer.passHostHeader=true

volumes:
  django_app_data:
    external: true
    name: django_app_data
  nginx_data:
    external: true
    name: nginx_data

networks:
  network_public:
    name: network_public
    external: true